ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleManager.java

```diff
@@ -151,73 +122,63 @@ boolean isAdPlayable(@NonNull String placement, @NonNull AdSize adSize) {
     /**
      * Checks and returns if the passed Placement ID is a valid placement for App ID
      *
-     * @param placementId
+     * @param placementId placement identifier
      * @return
      */
     boolean isValidPlacement(String placementId) {
-        return Vungle.isInitialized() &&
-                Vungle.getValidPlacements().contains(placementId);
+        return Vungle.isInitialized() && Vungle.getValidPlacements().contains(placementId);
     }
 
-    VungleBanner getVungleBanner(String placement, AdConfig.AdSize adSize, VungleListener vungleListener) {
-        Log.d(TAG, "getVungleBanner");
-        VungleBanner bannerAd = Banners.getBanner(placement, adSize, playAdCallback(vungleListener));
-        if (bannerAd != null) {
-            activeBannerAds.put(placement, bannerAd);
-        }
-
-        return bannerAd;
-    }
-
-    VungleNativeAd getVungleNativeAd(String placement, AdConfig adConfig,
-                                     VungleListener vungleListener) {
-        Log.d(TAG, "getVungleNativeAd");
-
-        //Fetch new ad
-        VungleNativeAd bannerAd = Vungle.getNativeAd(placement, adConfig,
-                playAdCallback(vungleListener));
-        if (bannerAd != null) {
-            activeNativeAds.put(placement, bannerAd);
+    @Nullable
+    synchronized VungleBannerAdapter getBannerRequest(@NonNull String placementId, @Nullable String requestUniqueId) {
+        VungleBannerAdapter bannerRequest = mVungleBanners.get(placementId);
+        if (bannerRequest != null) {
+            String activeUniqueRequestId = bannerRequest.getUniquePubRequestId();
+            if (bannerRequest.isLoading()) {
```

Comment:
why new refresh can't adopt the loading request?

Reply:
try to throw away the second request because if in case of both uniqueReqestIds are the same or null we can't identify refresh or new request.

Reply:
removed this check: if both the same let it go.